tests/heatbath_test.cpp

```diff
@@ -249,12 +251,30 @@ int main(int argc, char **argv)
 
     loadGaugeQuda(gauge->Gauge_p(), &gauge_param);
     double3 plaq = plaquette(*gaugeEx);
-    double charge = qChargeQuda();
-    printfQuda("Initial gauge field plaquette = %e topological charge = %e\n", plaq.x, charge);
+
+    GaugeFieldParam tensorParam(gaugeEx->X(), gaugeEx->Precision(), QUDA_RECONSTRUCT_NO, 0, QUDA_TENSOR_GEOMETRY);
+    tensorParam.siteSubset = QUDA_FULL_SITE_SUBSET;
+    tensorParam.order = QUDA_FLOAT2_GAUGE_ORDER;
+    tensorParam.ghostExchange = QUDA_GHOST_EXCHANGE_NO;
+    cudaGaugeField Fmunu(tensorParam);
+
+    computeFmunu(Fmunu, *gauge);
+    double charge = quda::computeQCharge(Fmunu);
+    // double charge = qChargeQuda();
+    printfQuda("Initial gauge field plaquette = %.16e topological charge = %.16e\n", plaq.x, charge);
 
     // Reunitarization setup
     setReunitarizationConsts();
-    plaquette(*gaugeEx);
+    // CallUnitarizeLinks(gaugeEx);
```

Comment:
What's the motivation for adding these commented out calls?

Reply:
MILC performs a reunitarisation as standard before the Wilson Flow routine is called. I thought the discrepancies were coming from that, but it transpired not to be the case as even further data divergence was seen, even in the energy. I suspected I was not using it properly. It would perhaps be a good idea to allow the user to call reunitarisation for any QUDA test, orthogonal to this PR.